src/main/scala/io/iohk/ethereum/ledger/BlockExecution.scala

```diff
@@ -59,7 +60,9 @@ class BlockExecution(
         .getBlockHeaderByHash(block.header.parentHash)
         .toRight(MissingParentError) // Should not never occur because validated earlier
       execResult <- executeBlockTransactions(block, parent)
-      worldToPersist = blockPreparator.payBlockReward(block, execResult.worldState)
+      worldToPersist <- Either
+        .catchOnly[Throwable](blockPreparator.payBlockReward(block, execResult.worldState))
```

Comment:
Now with cats making it easy to specify the error to catch, I'm thinking if we should specialize this to `MPTException`.

Reply:
you tell me. Is it possible other exceptions are thrown while executing payBlockReward?

Reply:
Perhaps but not at first glance. I propose to be more conservative and only capture the ones we know about now (so `MPTException`.)